examples/deployer/src/index.ts

```diff
@@ -38,18 +38,19 @@ async function main() {
 
   // Connect it with a Provider (https://docs.ethers.org/v5/api/providers/)
   const network = "goerli";
-  const provider = new ethers.providers.InfuraProvider(network);
-  const connectedSigner = turnkeySigner.connect(provider);
+  const connectedSigner = turnkeySigner.connect(
+    new ethers.InfuraProvider(network)
+  );
 
-  const chainId = await connectedSigner.getChainId();
+  const chainId = (await connectedSigner.provider?.getNetwork())?.chainId ?? 0;
```

Comment:
i feel like we could make provider a hard requirement (`.provider!`)

Reply:
Probably not as important for examples but I think best practice is to only use `!` when you know that the value can’t be null or undefined. But I wonder if the types should be adjusted in the signer so that provider isn't undefined? 

```
constructor(config: TConfig, provider?: Provider) {
    super(provider);

    this.client = config.client;
    this.organizationId = config.organizationId;
    this.signWith = config.signWith;
  }

  connect(provider: Provider): TurnkeySigner {
    return new TurnkeySigner(
      {
        client: this.client,
        organizationId: this.organizationId,
        signWith: this.signWith,
      },
      provider
    );
  }
```

It wasn't totally clear to me the reason for having the `connect` method and the constructor. But if we made the provider a hard requirement in the constructor then users wouldn't need to check if the provider is defined on the signer before using it. Is that change possible? Or am I missing context on the usage of the `connect` method?